R/mergeData.R

```diff
@@ -0,0 +1,24 @@
+#' @title Incorporating feature engineering with clinical data
+#'
+#' @export
+
+mergeData <- function(feat_eng_df, id_mapping = NULL, test_clinical_data) {
+
+  # note: still need to generalize this to more colnames.
+  if (!is.null(id_mapping)) {
+  colnames(id_mapping) <- c("PatID", "pid")
+  df_merged <- left_join(feat_eng_df, id_mapping, by = "pid") %>%
+    dplyr::relocate(PatID, .after = "pid") %>%
+    dplyr::select(-pid)
+  result <- left_join(df_merged, test_clinical_data, by = "PatID") %>%
+    na.omit()
```

Comment:
Because when we `left_join` the `df_merged` tibble with `test_clinical_data`, there could be identification values that don't match, and those will display NA in the first column.

Reply:
The problem is that na.omit() removes the entire row if there is a single NA. Some of the clinical entries are simply missing. Maybe you can randomly impute some NAs into the test data and see if the code still works

Reply:
I think you can specify columns to remove in na.omit() though, so it doesn't remove the whole row

Reply:
I got it, the function `tidyr::drop_na` works perfectly for this. I've set it so that if any values in the `pid` column are NA, then the row will be dropped, but it won't be dropped if any of the clinical columns are NA.